scrc/dataset_creation/dataset_creator.py

```diff
@@ -266,20 +270,100 @@ def get_dataset_folder(self):
             return self.create_dir(self.tmp_subdir, self.dataset_name)
         return self.create_dir(self.datasets_subdir, self.dataset_name)
 
-    def create_dataset(self, sub_datasets=False, kaggle=False, save_reports=False):
+    def create_dataset(self, court_list=None, concatenate=False, sub_datasets=False, kaggle=False, save_reports=False):
         """
         Retrieves the respective function named by the dataset and executes it to get the df for that dataset.
         :return:
         """
+        if court_list is None:
+            court_list = ["CH_BGer"]  # default to BGer
+
         self.logger.info(f"Creating {self.dataset_name} dataset")
 
         # TODO in the future: maybe save text as list of paragraphs
         # TODO make sure that the same data is saved to kaggle, csv and huggingface format!
 
-        dataset, labels = self.prepare_dataset(save_reports)
-        dataset = dataset.shuffle(seed=42)
-        self.save_dataset(dataset, labels, self.get_dataset_folder(), self.split_type,
-                          sub_datasets=sub_datasets, kaggle=kaggle, save_reports=save_reports)
+        not_created = []
+        created = []
+
+        datasets_list = []
+        labels_list = []
+
+        for court_string in court_list:
+            self.logger.info(f"Creating dataset for {court_string}")
+            dataset, labels = self.prepare_dataset(save_reports, court_string=court_string)
+
+            if len(dataset) == 0:
+                self.logger.info(f"Dataset for {court_string} could not be created")
+                not_created.append(court_string)
+            else:
+                dataset = dataset.shuffle(seed=42)
```

Comment:
on second thought, I think we can remove this line. When training, we usually shuffle the data anyway again.

Reply:
Alright, makes the code a little bit easier to read too.